src/libs/Network.js

```diff
@@ -53,9 +67,14 @@ function processNetworkRequestQueue() {
             return;
         }
 
+        const requestData = queuedRequest.data;
+        if (email) {
+            requestData.email = email;
+        }
+
         const finalParameters = _.isFunction(enhanceParameters)
-            ? enhanceParameters(queuedRequest.command, queuedRequest.data)
-            : queuedRequest.data;
+            ? enhanceParameters(queuedRequest.command, requestData)
+            : requestData;
```

Comment:
NAB, might make more sense to add the email to the `finalParameters` if we have one. Then we do not need `requestData`.